test/__mocks__/@npmcli/arborist.js

```diff
@@ -0,0 +1,3 @@
+const arb = jest.genMockFromModule('@npmcli/arborist');
```

Comment:
I think this `__mocks__` directory should be placed in the root directory, next to `node_modules`: https://jestjs.io/docs/en/manual-mocks#mocking-node-modules. Maybe that's why you can't manage to mock the module?

Reply:
the thing is that mocking module works well, but the class that I mock, doesn't contain property that is later exposed in the class (this property is added after `reify()`. So I have to somehow extend mock with additional mocked property, and this is where I fail. I mean I am able to add props, and when I console log the module inside test, the function is there. Thing is that it is lost inside the code that is tested because `new` (in the code) overrides what I add in the test . I hope that makes sense

Reply:
I see. There are two things that come to my mind here:

1. Why don't you create the `resolvedAdd` mock here in this file? This way, it should be available without having to explicitly do it on the test.
2. When you do `new`, what you get is the result of "copying" the `prototype` of the object. So probably, what you have to do is to mock `arb.prototype.resolvedAdd` instead of `arb.resolvedAdd`.

Does it make sense?

Reply:
> When you do new, what you get is the result of "copying" the prototype of the object

omg, fundamental knowledge 🤦🏼 
I love you mate! ❤️  tests are green like fields of Ireland ☘️